server/src/main/java/org/opensearch/search/aggregations/metrics/CardinalityAggregatorFactory.java

```diff
@@ -74,7 +104,7 @@ public static void registerAggregators(ValuesSourceRegistry.Builder builder) {
 
     @Override
     protected Aggregator createUnmapped(SearchContext searchContext, Aggregator parent, Map<String, Object> metadata) throws IOException {
-        return new CardinalityAggregator(name, config, precision(), searchContext, parent, metadata);
```

Comment:
I would move `executionMode` to last since its least critical and optional parameter and newly introduced.